### DISCUSSION TITLE
scorecard unable to connect to https://review.trustedfirmware.org/TF-M/trusted-firmware-m

### DISCUSSION THREAD

**Discussion by pradyumna212** [ossf/scorecard#2481] [Q&A] [answered]
I am unable to use Scorecard on non-Github (gerrit and gitlab explicitly) repos.. Can anyone help please? 

# scorecard --show-details --format json --repo=https://review.trustedfirmware.org/TF-M/trusted-firmware-m

Error: GetClients: getting local directory client: error in IsValid: unsupported host: [review.trustedfirmware.org](http://review.trustedfirmware.org/)
2022/11/23 14:48:25 error during command execution: GetClients: getting local directory client: error in IsValid: unsupported host: [review.trustedfirmware.org](http://review.trustedfirmware.org/)

**Comment by spencerschrock** [comment, marked as answer]
If you're providing a URL via the `--repo` argument, `Scorecard` only supports GitHub (and GitLab in the works). 
You would want to clone the repo locally and try running with the `--local` flag.

Unfortunately not every check works locally, and not all of the ones that do are useful outside a GitHub/GitLab context.

```
git clone https://git.trustedfirmware.org/TF-M/trusted-firmware-m.git/
scorecard --local=trusted-firmware-m --show-details --format json
```
